database/behavior/thumbnailable.php

```diff
@@ -15,11 +15,57 @@
  */
 class ComFilesDatabaseBehaviorThumbnailable extends KDatabaseBehaviorAbstract
 {
+    /**
+     * @var array A list of files extensions for which thumbnails may be generated
+     */
+    protected $_thumbnailable_extensions;
+
+    public function __construct(KObjectConfig $config)
+    {
+        parent::__construct($config);
+
+        $this->_thumbnailable_extensions = KObjectConfig::unbox($config->thumbnailable_extensions);
+    }
+
+    protected function _initialize(KObjectConfig $config)
+    {
+        $config->append(array('thumbnailable_extensions' => ComFilesModelEntityFile::$image_extensions));
+
+        parent::_initialize($config);
+    }
+
+    /**
+     * Tells if a file contains an image
+     *
+     * Files containing images are thumbnailable, meaning that a thumbnail may be generated for them
+     *
+     * @return bool true if it contains an image, false otherwise
+     */
+    public function containsImage()
```

Comment:
@amazeika shall we rename this method to something clearer? Something like `canGenerateThumbnail` ? Right now it's not immediately obvious to me what it does

Reply:
@ercanozkaya How about thumbnailable()? ... the condition would be if ($file->thumbnailable()) ... another suggestion would be $file->hasImage() ... thoughts?

Reply:
@amazeika I think it would be confusing since isThumbnailable is already injected. Not sure about hasImage since you don't have an image yet, you can just have one in the future.

Reply:
@ercanozkaya isXXX is not possible indeed. What I would like is a method that makes sense when calling it against the file entity :)

Reply:
@amazeika Decided on `$file->canHaveThumbnail`  through slack